### ISSUE TITLE
A user must be notified when a ticket from a friend arrives in their wallet

### ISSUE THREAD

**Issue by zmelamed** [big-neon/bn-api#181]
child of big-neon/docs#21
connected to big-neon/bigneon#48

**Comment by Timo614** [comment]
I think this may already exist (at least on the bn-api side) via: https://github.com/big-neon/bn-api/pull/1124

**Comment by Timo614** [comment]
From my reading of documents on expo I see it's something to help build native apps so likely can't be leveraged here / it's completely different than a socket based solution (was thinking it was some sort of central server that handled the socket traffic between the web app and the API on first glance).

@zmelamed can you reply about what sort of notifications you were thinking here? Did you want to add some sort of socket.io solution for bn-web to alert for updates as well? Or just an email for this case?

**Comment by Krakaw** [comment]
We have the push notifications system in place, `api/src/domain_events/executors/broadcast_push_notification.rs`
We could add a new `BroadcastType` for the Broadcasts system `db/src/models/broadcasts.rs`.
This would be step one, we would then add the client side experience in the app

**Comment by Timo614** [comment]
Sounds good. Just reconfirming based on above this is for adding those push notifications? Not for an email?

Wasn't sure from the initial description initially if it was to support adding something for direct transfers communication, etc.

**Comment by Timo614** [comment]
@Krakaw I meant to chime in Friday but forgot to swing back here. I went ahead and started adding the new notification for that BroadcastType `TicketTransferReceived` and BroadcastAudience `User` but realized I'm effectively just recreating https://github.com/big-neon/bn-api/blob/master/api/src/communications/pushers/tickets_received.rs#L23 in a more complicated way. The broadcast push logic appears to normally just wrap the same sort of push communication https://github.com/big-neon/bn-api/blob/14d5059e39e796ac28082171a93046a0657e0d57/api/src/domain_events/executors/broadcast_push_notification.rs#L67 but in this case there's little value in doing so as the audience and type are just used as a means to trigger the push to that one user (which is the same as what is happening via that other method I linked above getting their push tokens etc).

I don't think there's anything on the bn-api side to do here then as we have the existing push already happening via the direct transfers. Unless I'm missing something about what you wrote above. I'll stash my changes on a local branch in case we want to push them up but not sure if there's any value in that over just hitting the pusher tickets_received method like it's doing now.
